[NRFConnect] CI file dependencies are too narrow

### Reproduction steps

https://github.com/project-chip/connectedhomeip/actions/runs/15291664420/job/43012135386 is a failure on master of a PR that passed CI at https://github.com/project-chip/connectedhomeip/actions/runs/15285268337/job/42993849742

But on the PR, the thing that fails on master was skipped, because the test YAML (examples-nrfconnect.yaml) has:

```
            - name: Run unit tests for Zephyr native_posix_64 platform
              if: github.event_name == 'push' || steps.changed_paths.outputs.tests == 'true' || steps.changed_paths.outputs.nrfconnect == 'true'
```

and in this case what changed is pigweed, but in a way that breaks NRFConnect but _only_ NRFConnect apparently:

```
INFO    In file included from /__w/connectedhomeip/connectedhomeip/third_party/pigweed/repo/pw_string/public/pw_string/string_builder.h:33,
INFO                     from /__w/connectedhomeip/connectedhomeip/third_party/pigweed/repo/pw_unit_test/light_public_overrides/pw_unit_test/framework_backend.h:36,
INFO                     from /__w/connectedhomeip/connectedhomeip/third_party/pigweed/repo/pw_unit_test/public/pw_unit_test/framework.h:16,
INFO                     from /__w/connectedhomeip/connectedhomeip/src/test_driver/nrfconnect/main/runner.cpp:21:
INFO    /__w/connectedhomeip/connectedhomeip/third_party/pigweed/repo/pw_string/public/pw_string/string.h:27:10: fatal error: pw_containers/internal/traits.h: No such file or directory
INFO       27 | #include "pw_containers/internal/traits.h"
INFO          |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

### Platform

nrf connect

### Platform Version(s)

_No response_

### Type

CI tested

### (Optional) If manually tested please explain why this is only manually tested

_No response_

### Anything else?

_No response_

--- a/src/test_driver/nrfconnect/CMakeLists.txt
+++ b/src/test_driver/nrfconnect/CMakeLists.txt
@@ -74,6 +74,7 @@ target_include_directories(app PUBLIC
   ${PIGWEED_ROOT}/pw_string/public
   ${PIGWEED_ROOT}/pw_unit_test/public
   ${PIGWEED_ROOT}/pw_unit_test/light_public_overrides
+  ${PIGWEED_ROOT}/pw_containers/public
   ${PIGWEED_ROOT}/third_party/fuchsia/repo/sdk/lib/stdcompat/include
 )
 target_link_libraries(app PUBLIC chip $<TARGET_FILE:kernel>)

